gradle/project.gradle.kts

```diff
@@ -12,29 +16,28 @@ import kotlin.text.Charsets.UTF_8
 
 // Note: the plugins block is evaluated before the script itself, so no variables can be used
 plugins {
-    kotlin("jvm") version "1.2.31"
+    kotlin("jvm") version "1.2.40"
     application
     id("com.github.ben-manes.versions") version "0.17.0"
+    id("de.undercouch.download") version "3.4.2"
 }
 
 /** working directory of the application */
 val workDir = file("working dir")
-val javaVersion = JavaVersion.VERSION_1_9
 val kotlinVersion: String by extra {
     buildscript.configurations["classpath"]
             .resolvedConfiguration.firstLevelModuleDependencies
             .find { it.moduleName=="org.jetbrains.kotlin.jvm.gradle.plugin" }!!.moduleVersion
 }
+val supportedJavaVersions = arrayOf(JavaVersion.VERSION_1_9, JavaVersion.VERSION_1_10)
 
-if (JavaVersion.current()!=javaVersion) {
+if (JavaVersion.current() !in supportedJavaVersions) {
     println("""org.gradle.java.home=${properties["org.gradle.java.home"]}
-		|Java version ${JavaVersion.current()} can't be used. Set $javaVersion as system default or create a "gradle.properties" file with "org.gradle.java.home" pointing to JDK $javaVersion""".trimMargin())
+		|Java version ${JavaVersion.current()} can't be used. Set one of ${supportedJavaVersions.joinToString()} as system default or create a "gradle.properties" file with "org.gradle.java.home" pointing to a supported Java version""".trimMargin())
     throw IllegalStateException("Invalid Java version: ${JavaVersion.current()}")
 }
 
 java {
-    targetCompatibility = javaVersion
```

Comment:
Why was this deleted?

Reply:
because sourceCompatibility and targetCompatibility are implicitly the latest version